Dockerfile

```diff
@@ -37,3 +37,11 @@ RUN make EXPERIMENTAL=${EXPERIMENTAL} cross
 FROM cross AS e2e-cross
 ARG EXPERIMENTAL="off"
 RUN make EXPERIMENTAL=${EXPERIMENTAL} e2e-cross
+
+# Image to be pushed to DockerHub
+# To be run as:
+# docker run -v ${PWD}:/mnt -w /mnt docker/docker-app --help
+FROM alpine AS export-image
+COPY --from=cross /go/src/github.com/docker/app/bin/docker-app-linux /usr/bin
```

Comment:
What about having an image with the 3 binaries (windows/mac/linux) so people can fetch docker-app though DockerHub, and adding something like this in the documentation:
```
This command will copy the docker-app executable to /usr/local/bin:
$ docker pull docker/app:latest &&
    id=$(docker create docker/app:latest) &&
    sudo docker cp $id:/docker-app-$(uname -s)-$(uname -m) /usr/local/bin/docker-app &&
    (docker rm $id >/dev/null)
```
Any thought @garethr ? cc @simonferquel @chris-crone @vdemeester ?

Reply:
@silvin-lubecki isn't the PR meant to have a *runnable* image (and thus only need, linux for now :sweat_smile:) ? People can get binaries (ou should be able to) using the releases pages (where all supported target would be).

Reply:
I think that makes determining intent harder when it comes to pulls of the image. Distribution will be via the Docker packages eventually. For now via GitHub works well. I don't think I've seen this pattern elsewhere.

Note however we should have a multi-platform image, so we have a Windows container variant of this as well. Happy for that to come in a separate PR though.

Reply:
Added #421